spotlight-client/src/contentModels/Metric.ts

```diff
@@ -85,34 +70,39 @@ export default class Metric<RecordFormat extends AnyRecord> {
   readonly tenantId: TenantId;
 
   // data properties
-  private readonly dataTransformer: DataTransformer<RecordFormat>;
+  protected dataTransformer: (d: RawMetricData) => RecordFormat[];
 
-  private readonly sourceFileName: string;
+  protected readonly sourceFileName: string;
 
   isLoading?: boolean;
 
-  private allRecords?: RecordFormat[];
+  allRecords?: RecordFormat[];
 
   error?: Error;
 
+  // filter properties
+  localityId: RecordFormat extends LocalityFields ? string : undefined;
+
+  demographicView: RecordFormat extends DemographicFields
+    ? DemographicView
+    : undefined;
+
   constructor({
     name,
     description,
     methodology,
     tenantId,
-    dataTransformer,
     sourceFileName,
-  }: InitOptions<RecordFormat>) {
-    makeAutoObservable(this, {
-      // readonly properties do not need to be observed
-      description: false,
-      methodology: false,
-      name: false,
-      tenantId: false,
-      // in practice, collections should not change once we are done
-      // bootstrapping them (which is done right after instantiation);
-      // no need to make them observable
-      collections: false,
+    dataTransformer,
+    defaultDemographicView,
+    defaultLocalityId,
+  }: BaseMetricConstructorOptions<RecordFormat>) {
+    makeObservable(this, {
```

Comment:
apparently you can't use `makeAutoObservable` on classes that are part of an inheritance tree, I guess it can't reliably figure out what all the members will be? It's a little annoying but it makes sense. Mobx threw a fatal error alerting me to this fact